main.js

```diff
@@ -322,11 +322,13 @@ function createStorinfoClient(cfg, clients, barrier) {
         standalone: false
     };
 
-    clients.storinfo = storinfo.createClient(opts);
+    var siClient = storinfo.createClient(opts);
 
-    clients.storinfo.once('topology', function () {
+    siClient.once('topology', function () {
         opts.log.info('first poll completed');
-        barrier.done('createStorinfoClient');
+        // Intentionally only assign `clients.storinfo` after it is intialized
+        // so its presence can be used to test if init is complete.
+        clients.storinfo = siClient;
     });
```

Comment:
There are two relevant changes to storinfo setup here:

1. we only set `clients.storinfo` after it is initialized so that the "ping" handler above can use that as the signal that we are ready to handle writes.
2. We drop using the "barrier" to signal that the storinfo client is ready. Lower in main.js it states that this "barrier" is for:

```
    /*
     * Create a barrier to ensure client connections that are established
     * asynchronously and are required for muskie to serve a minimal subset of
     * requests are ready prior to starting up the restify servers.
     */
```

We actually want to start restify listening before storinfo is initialized so that (a) we can handle "GET /ping" that haproxy is calling (we'll return 503 from that) and (b) theoretically so reads can still be served (though haproxy won't be adding this endpoint to the rotation).